packages/domain/lib/src/use_cases/generate_guess_number.dart

```diff
@@ -0,0 +1,16 @@
+import 'dart:math';
+import 'package:domain/src/use_cases/base/base_use_case_out.dart';
+
+const maxGuessNumber = 10;
+
+class GenerateGuessNumberUseCase extends BaseUseCaseOut<int> {
+  @override
+  int call() => _generateGuessNumber();
```

Comment:
(up to u) no need in extracting this logic into separate method if u then call only it in another method. u create redundant nesting